### PR TITLE
Move to MySQL 8.0 as the default template generation (#11153)

### PR DESCRIPTION
**Pull request by rsajwani**
* Move towards MySQL 8.0 as the default template generation

This upgrades the remaining things to Ubuntu 20.04 and makes MySQL 8.0 the default we run tests against. We still have tests for MySQL 5.7 but those are now explicitly opted into.

This should finish up the Ubuntu 20.04 upgrade and also makes things easier for the future when we need to upgrade again.

Signed-off-by: Dirkjan Bussink <d.bussink@gmail.com>

* CI: rename shard vtorc_8.0 to vtorc_5.7, change expected test output for 8.0

Signed-off-by: deepthi <deepthi@planetscale.com>

* CI: increase timeout for 8.0 tests on the actual test step from 30 to 45 mins

Signed-off-by: deepthi <deepthi@planetscale.com>

* CI: increase timeout to 45 minutes for mysql57 tests too. We really only need this for vtorc, but I've made the change to the template so all tests get it.

Signed-off-by: deepthi <deepthi@planetscale.com>

* CI: fix vtorc test to work with both 5.7 and 8.0

Signed-off-by: deepthi <deepthi@planetscale.com>

* CI: missed docker flag in mysql57 template, one more fix to vtorc test

Signed-off-by: deepthi <deepthi@planetscale.com>

* removing spaces from pb file

Signed-off-by: Rameez Sajwani <rameezwazirali@hotmail.com>

* removing spaces in pb file part 2

Signed-off-by: Rameez Sajwani <rameezwazirali@hotmail.com>

Signed-off-by: Dirkjan Bussink <d.bussink@gmail.com>
Signed-off-by: deepthi <deepthi@planetscale.com>
Signed-off-by: Rameez Sajwani <rameezwazirali@hotmail.com>
Co-authored-by: deepthi <deepthi@planetscale.com>
Co-authored-by: Rameez Sajwani <rameezwazirali@hotmail.com>

## Description

There are two main changes in CI workflow been done as part of this PR.
- Move to ubuntu 20.04
- Upgrade Mysql 5.7 to Mysql 8.0.31 (latest)

This PR cherry-pick (#11153) the change from main where we have made Mysql8.0.x as default database (from Mysql 5.7). We used the same approach as used in 'main' which is using mysql-apt-config to install mysql server. Currently it is 8.0.31.

A lot of vreplication tests were failing. The reason for failure was the machine was running out of resources. We change innodb_buffer_pool_size = 16M in order to get around that.

## Related Issue(s)
close https://github.com/vitessio/vitess/issues/11511

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: .github/docker/cluster_test_vreplication_v2/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# DO NOT MODIFY: THIS FILE IS GENERATED USING "make generate_ci_workflows"
+
+ARG bootstrap_version=10.2
+ARG image="vitess/bootstrap:${bootstrap_version}-mysql80"
+
+FROM "${image}"
+
+USER root
```
Comment (GuptaManan100) [inline .github/docker/cluster_test_vreplication_v2/Dockerfile:8]: Is this file still required?

File: go/test/endtoend/backup/mysqlctld/backup_mysqlctld_test.go [hunk 2]
```diff
@@ -24,5 +24,5 @@ import (
 
 // TestBackupMysqlctld - tests the backup using mysqlctld.
 func TestBackupMysqlctld(t *testing.T) {
-	backup.TestBackup(t, backup.Mysqlctld, "", 0)
+	backup.TestBackup(t, backup.Mysqlctld, "xbstream", 0)
```
Comment (GuptaManan100) [inline go/test/endtoend/backup/mysqlctld/backup_mysqlctld_test.go:27]: I am not sure about this change either.

File: go/vt/mysqlctl/mysqld.go [hunk 3]
```diff
@@ -165,6 +165,7 @@ func NewMysqld(dbcfgs *dbconfigs.DBConfigs) *Mysqld {
 				os.Getenv("VTROOT"),
 				vtenvMysqlRoot,
 				os.Getenv("MYSQL_FLAVOR"))
+			log.Errorf(message)
```
Comment (GuptaManan100) [inline go/vt/mysqlctl/mysqld.go:168]: errant Log message or do we want to commit this?

File: test/ci_workflow_gen.go [hunk 4]
```diff
@@ -121,16 +120,17 @@ var (
...
 		"vreplication_multicell",
 		"vreplication_cellalias",
 		"vreplication_basic",
-		"vreplication_v2",
 		"vtorc",
 		"vtorc_8.0",
 		"schemadiff_vrepl",
 		"topo_connection_cache",
+		"vreplication_v2",
 	}
```
Comment (GuptaManan100) [inline test/ci_workflow_gen.go:135]: The ordering of `vreplication_v2` was correct before

Comment (GuptaManan100) [reply test/ci_workflow_gen.go:135]: Actually no, the vtorc one are also wrong

**Review by rohit-nayak-ps — APPROVED** [review]
vrep changes lgtm

File: .github/docker/cluster_test_vreplication_v2/Dockerfile (hunk 1 above)
Comment (rsajwani) [reply .github/docker/cluster_test_vreplication_v2/Dockerfile:8]: deleted them..

File: go/test/endtoend/backup/mysqlctld/backup_mysqlctld_test.go (hunk 2 above)
Comment (rsajwani) [reply go/test/endtoend/backup/mysqlctld/backup_mysqlctld_test.go:27]: this is because xtrabackup 8.0 with mysql8.0 does not support "tar" (which is default) but xbstream. This change is cherrypicked from https://github.com/vitessio/vitess/pull/10691/files#diff-909528cb082669b2422729be9acff461e293f142b88072415a517977d87ec76e

File: go/vt/mysqlctl/mysqld.go (hunk 3 above)
Comment (rsajwani) [reply go/vt/mysqlctl/mysqld.go:168]: Yes we need it. I saw during startup if we don't able to find MySQL then error log does not have any information.

File: test/ci_workflow_gen.go (hunk 4 above)
Comment (rsajwani) [reply test/ci_workflow_gen.go:135]: so you want me to restore to original or not ??

Comment (rsajwani) [reply test/ci_workflow_gen.go:135]: moved it to original place

**Review by GuptaManan100 — APPROVED** [review]
LGTM!
